webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/BinaryDataService.scala

```diff
@@ -84,8 +87,10 @@ class BinaryDataService(val dataBaseDir: Path,
     if (request.dataLayer.doesContainBucket(bucket) && request.dataLayer.containsResolution(bucket.mag)) {
       val readInstruction =
         DataReadInstruction(dataBaseDir, request.dataSource, request.dataLayer, bucket, request.settings.version)
-      val bucketProvider = bucketProviderCache.getOrLoadAndPut(request.dataLayer)(dataLayer =>
-        dataLayer.bucketProvider(dataVaultServiceOpt))
+      val dataSourceId = if (request.dataSource != null) request.dataSource.id else DataSourceId("", "")
```

Comment:
When is this the case and should it lead to something else, since this will not use the right cache?

Reply:
I added an explaining comment “dataSource is null and unused for volume tracings. Insert dummy DataSourceId (also unused in that case)”

Related: https://github.com/scalableminds/webknossos/issues/6234